go/v1/resource_descriptor.go

```diff
@@ -16,16 +16,77 @@ var (
 	ErrRDRequiredField       = errors.New("at least one of name, URI, or digest are required")
 )
 
+type HashAlgorithm string
+
+const (
+	AlgorithmMD5        HashAlgorithm = "md5"
+	AlgorithmSHA1       HashAlgorithm = "sha1"
+	AlgorithmSHA224     HashAlgorithm = "sha224"
+	AlgorithmSHA512_224 HashAlgorithm = "sha512_224"
+	AlgorithmSHA256     HashAlgorithm = "sha256"
+	AlgorithmSHA512_256 HashAlgorithm = "sha512_256"
+	AlgorithmSHA384     HashAlgorithm = "sha384"
+	AlgorithmSHA512     HashAlgorithm = "sha512"
+	AlgorithmSHA3_224   HashAlgorithm = "sha3_224"
+	AlgorithmSHA3_256   HashAlgorithm = "sha3_256"
+	AlgorithmSHA3_384   HashAlgorithm = "sha3_384"
+	AlgorithmSHA3_512   HashAlgorithm = "sha3_512"
+	AlgorithmGitCommit  HashAlgorithm = "gitCommit"
```

Comment:
could this also include `gitTag`, `gitBlob`, `gitTree`?

Reply:
Sure thing! I've repushed the commit including them  :+1:

Reply:
thanks!